### PR TITLE
Swift 5 Migration With Backwards Compatibility

### PR DESCRIPTION
**Pull request by gtrop1**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,4 +1,35 @@
...
+  gem "rspec", "~> 3.8.0", require: false
+  gem "rspec_junit_formatter", "~> 0.4.1", require: false
+  gem "rubocop", "~> 0.67.2", require: false
+  gem "rubocop-performance", "~> 1.1.0", require: false
+  gem "simplecov", "~> 0.16.1", require: false
+  gem "simplecov-json", "~> 0.2.0", require: false
+end
+
+plugins_path = File.join(File.expand_path(__dir__), "Fastlane", "Pluginfile")
+eval_gemfile(plugins_path) if File.exist?(plugins_path)
```
Comment (cnoon) [inline Gemfile]: Needs copyright header update and should only be `cocoapods` and `xcpretty`. We don't need the others for this project.

File: Willow.podspec [hunk 2]
```diff
@@ -9,7 +9,7 @@ Pod::Spec.new do |s|
 
   s.source = { :git => "https://github.com/Nike-Inc/Willow.git", :tag => s.version }
   s.source_files = "Source/*.swift"
-  s.swift_version = "4.2"
+  s.swift_versions = ['4.2', '5.0']
```
Comment (cnoon) [inline Willow.podspec]: Can we switch over to using double instead of single quotes?

File: .travis.yml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 os: osx
-osx_image: xcode10
+osx_image: xcode10.2
```
Comment (cnoon) [inline .travis.yml:2]: Probably should bump the simulator versions as well in here.

File: CHANGELOG.md [hunk 4]
```diff
@@ -33,6 +33,8 @@ All notable changes to this project will be documented in this file.
 #### Added
 
 #### Updated
+- Upgraded to Swift 5 with backwards compatability with 4.2 and Xcode 10.1
+  - To be updated by [Greg Tropino](https://github.com/gtrop1) in Pull Request
```
Comment (cnoon) [inline CHANGELOG.md]: `Upgraded to` -> `To`. The `Updated` is assumed. And you can set the PR number now.

Comment (cnoon) [reply CHANGELOG.md]: Also, could you bump the version in the podspec and Info.plist?

**Review by cnoon — CHANGES_REQUESTED** [review]
A few small things. We'll also need to get the package files updated.

File: CHANGELOG.md (hunk 4 above)
Comment (gtrop1) [reply CHANGELOG.md]: Updated version in podspec and Info.plist, modified changelog per your suggestion.

File: .travis.yml (hunk 3 above)
Comment (gtrop1) [reply .travis.yml:2]: Simulators have been bumped up.

File: Willow.podspec (hunk 2 above)
Comment (gtrop1) [reply Willow.podspec]: Done!

File: Gemfile (hunk 1 above)
Comment (gtrop1) [reply Gemfile]: Modified copyright header to proper year and trimmed the gems down as suggested and re-installed to update the lockfile.

File: Package@swift-4.2.swift [hunk 5]
```diff
@@ -0,0 +1,38 @@
+// swift-tools-version:4.2
+// The swift-tools-version declares the minimum version of Swift required to build this package.
```
Comment (cnoon) [inline Package@swift-4.2.swift]: Could we ditch this comment now?

File: Package.swift [hunk 6]
```diff
@@ -28,11 +27,21 @@ import PackageDescription
...
-        .library(name: "Willow", targets: ["Willow"])
+        .library(
+            name: "Willow",
+            targets: ["Willow"])
     ],
     targets: [
-        .target(name: "Willow", path: "", sources: ["Source"])
+        .target(
+            name: "Willow",
+            path: "Source")
```
Comment (cnoon) [inline Package.swift]: Could one-line these if you wanted.

File: Gemfile.lock [hunk 7]
```diff
@@ -56,25 +56,27 @@ GEM
...
+      xcpretty (~> 0.2, >= 0.0.7)
 
 PLATFORMS
   ruby
 
 DEPENDENCIES
-  cocoapods (~> 1.6.0.beta.1)
-  xcpretty
+  cocoapods (~> 1.7.0.beta.3)
+  xcpretty-json-formatter (~> 0.1.0)
```
Comment (cnoon) [inline Gemfile.lock]: Should be `xcpretty`.

File: Gemfile [hunk 8]
```diff
@@ -1,4 +1,12 @@
+#
+#  Gemfile
+#  BuildTools
```
Comment (cnoon) [inline Gemfile]: `BuildTools` -> `Willow`?

**Review by cnoon — CHANGES_REQUESTED** [review]
A couple more things.

File: Package@swift-4.2.swift (hunk 5 above)
Comment (gtrop1) [reply Package@swift-4.2.swift]: Removed

File: Package.swift (hunk 6 above)
Comment (gtrop1) [reply Package.swift]: Changed to one line

File: Gemfile.lock (hunk 7 above)
Comment (gtrop1) [reply Gemfile.lock]: Set back to xcpretty and re-ran bundle install

File: Gemfile (hunk 8 above)
Comment (gtrop1) [reply Gemfile]: Oops! Corrected 😅

**Pull request merged**